### PR TITLE
Develop tab glt positioning

### PR DESCRIPTION
**Pull request by paul-bayleaf**
As per E-laute project implements positioning for German lute tablature using loc and valign.  Implements notationsubtype for different variants of the 6th course - although there is much more to do here.   Fix due to change in GetListSize interface.   Fix to use explicit strikethrough of digit one for open 6th course etc instead of SMUFL_E595_ornamentLeftVerticalStrokeWithCross.

### CODE REVIEW

**Comment by paul-bayleaf** [comment]
Fixes Issue #3473

**Comment by lpugin** [comment]
Thanks. Do we have a corresponding PR that has been made (and merged?) to the tablature-ig customization? We need to make sure the what comes into Verovio is in sync with that. Pinging @reinierdevalk @musicog @DILewis

**Comment by paul-bayleaf** [comment]
Thanks.   

I've updated my pull request for the MEI 4.0.1 schema customizations for the German lute tablature work.  I note that it also works for MEI 5.0.  I use it to verify the MEI output of my luteconv program.
https://github.com/music-encoding/tablature-ig/pull/26
But this has not been merged.

If you are referring to 
https://github.com/music-encoding/music-encoding/tree/tablature-ig
then that was last updated on 2021-11-16.  I've not touched this.

**Comment by lpugin** [comment]
Great, thanks. @reinierdevalk or @DILewis: could you look at updating the [tag](https://github.com/music-encoding/music-encoding/tree/tablature-ig) to MEI 5.0 ?

**Comment by lpugin** [comment]
I have not updated https://github.com/music-encoding/music-encoding/tree/tablature-ig and brought it inline with https://github.com/music-encoding/music-encoding/tree/develop.

The proposed changed should be added there. The two modifications adding `att.verticalAlign` to `note` and `staffDef` should be brought up for discussion since these are not tablature-specific elements.

File: src/note.cpp [hunk 1]
```diff
@@ -350,23 +352,38 @@ std::u32string Note::GetTabFretString(data_NOTATIONTYPE notationType, bool &over
...
+        // parts of the SMuFL collection.  Overlines and strike throughs are drawn separately.
 
-        if (course == 6 && fret >= 0 && fret <= 13) {
-            if (fret == 0) {
-                fretStr = SMUFL_E595_ornamentLeftVerticalStrokeWithCross; // substitute for 1 with oblique stroke
-            }
-            else {
+        if (course >= 6 && fret >= 0 && fret <= 13) {
+            // TODO more GLT subtypes needed, is this subtype too specific?
+            if (notationSubtype == "judenkunig_1523") {
```
Comment (lpugin) [inline src/note.cpp:368]: Looking again at this PR, this is also problematic. We should not have in Verovio hardcoded values like that. For the glyphs missing I would suggest using a symbol table. This is already supported by Verovio (see the test suite examples for symbols). Pinging @musicog.

Comment (musicog) [reply src/note.cpp:368]: As decided at the MEI Development Workshop at mdw, we will be defaulting to Newsidler unless an alternate symbol is defined (via `@glyphnum` or `@altsym` on `<note>`, or via a `<symbolTable>`)

Comment (rettinghaus) [reply src/note.cpp:368]: Just for reference: https://github.com/w3c/smufl/issues/274

Comment (paul-bayleaf) [reply src/note.cpp:368]: Does this mean that @notationsubtype is no longer to be used for the >= 6th course variants?   I.e. there are no "predefined variant sets"?

If so I can remove the @notationsubtype code and just keep the Newsidler variant.   This pull request would then  just implement the vertical positioning  with @loc and @valign and fix Issue https://github.com/rism-digital/verovio/issues/3473.   Would that be sufficient to have this pull request accepted?  Or are more changes required?

I would like to deal with implementing @glyphnum and @altsym etc in a separate pull request - first I need to study how to do this.

Thanks

Comment (lpugin) [reply src/note.cpp:368]: @paul-bayleaf yes please remove the `@notationsubtype`-dependent code and I can then merge the PR.

The `@valign` on `staffDef`will be renamed at some point, but the code can easily be adjusted once the renaming is done. Conceptually nothing will change so I am OK to merge it.

**Review by lpugin — APPROVED** [review]

**Pull request merged**